File: src/main/java/p00/Problem_0025.java
package p00;

import java.util.ArrayList;

public class Problem_0025 {

	@SuppressWarnings("unchecked")
	public static int solve() {

		ArrayList<Integer> a = new ArrayList<>();
		ArrayList<Integer> b = new ArrayList<>();
		a.add(1);
		b.add(1);

		int index = 2;

		while (b.size() < 1000) {

			index++;

			ArrayList<Integer> save = (ArrayList<Integer>) b.clone();
			add(b, a);
			a = save;

		}

		return index;
	}

	private static void add(ArrayList<Integer> pB, ArrayList<Integer> pA) {

		int help = 0;

		for (int i = 0; i < pA.size(); i++) {
			int number = pB.get(i) + pA.get(i) + help;
			help = number / 10;
			pB.set(i, number - help * 10);
		}

		for (int i = pA.size(); i < pB.size(); i++) {
			if (help == 0)
				break;
			int number = pB.get(i) + help;
			help = number / 10;
			pB.set(i, number - help * 10);
		}

		if (help > 0) {
			String s = "" + help;
			for (int i = 0; i < s.length(); i++) {
				pB.add(Integer.valueOf("" + s.charAt(i)));
			}
		}

	}

}

File: src/main/kotlin/p00/Problem0025.kt
package p00

class Problem0025 {
    fun solve(): Int {
        var a = ArrayList<Int>()
        val b = ArrayList<Int>()
        a.add(1)
        b.add(1)
        var index = 2
        while (b.size < 1000) {
            index++
            val save = b.toMutableList()
            add(b, a)
            a = save as ArrayList<Int>
        }
        return index
    }

    private fun add(pB: ArrayList<Int>, pA: ArrayList<Int>) {
        var help = 0
        for (i in pA.indices) {
            val number = pB[i] + pA[i] + help
            help = number / 10
            pB[i] = number - help * 10
        }
        for (i in pA.size until pB.size) {
            if (help == 0) break
            val number = pB[i] + help
            help = number / 10
            pB[i] = number - help * 10
        }
        if (help > 0) {
            val s = "" + help
            for (element in s) {
                pB.add(Integer.valueOf("" + element))
            }
        }
    }
}